// @mavric/apso-cli/dist/lib/utils/field.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.typeExistsInEntity = exports.fieldToEnumType = exports.getFieldForTemplate = exports.getDefaultValueForField = exports.getJsTypeFromFieldType = void 0;
const casing_1 = require("./casing");
const getJsTypeFromFieldType = (type) => {
    switch (type) {
        case "array":
            return "string[]";
        case "boolean":
            return "boolean";
        case "integer":
            return "number";
        case "float":
            return "number";
        case "text":
        case "enum":
        default:
            return "string";
    }
};
exports.getJsTypeFromFieldType = getJsTypeFromFieldType;
const getDefaultValueForField = (field) => {
    var _a, _b;
    switch (field.type) {
        case "enum":
            if (field.default === null) {
                return null;
            }
            if (typeof field.default !== "undefined") {
                if (!((_a = field.values) === null || _a === void 0 ? void 0 : _a.includes(field.default))) {
                    throw new Error(`${field.name}: Invalid default value '${field.default}' (Valid options are: '${(_b = field.values) === null || _b === void 0 ? void 0 : _b.join("', '")}')`);
                }
                return field.default;
            }
            if (typeof field.values !== "undefined") {
                return field.values[0];
            }
            return null;
        case "boolean":
            if (typeof field.default !== "undefined") {
                return field.default;
            }
            return "false";
        default:
            if (typeof field.default !== "undefined") {
                return field.default;
            }
            return null;
    }
};
exports.getDefaultValueForField = getDefaultValueForField;
const getFieldForTemplate = (fields, entityName) => fields.map((field) => ({
    ...field,
    default: (0, exports.getDefaultValueForField)(field),
    dataType: field.type === "enum"
        ? (0, exports.fieldToEnumType)(field.name, entityName)
        : (0, exports.getJsTypeFromFieldType)(field.type),
}));
exports.getFieldForTemplate = getFieldForTemplate;
const fieldToEnumType = (fieldName, entityName) => `${(0, casing_1.camelCase)(entityName, true)}${(0, casing_1.camelCase)(fieldName, true)}Enum`;
exports.fieldToEnumType = fieldToEnumType;
const typeExistsInEntity = (entity, type) => (entity === null || entity === void 0 ? void 0 : entity.fields)
    ? entity === null || entity === void 0 ? void 0 : entity.fields.findIndex((field) => field.type === type)
    : -1;
exports.typeExistsInEntity = typeExistsInEntity;
