// @mavric/apso-cli/dist/commands/server/new.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const core_1 = require("@oclif/core");
const child_process_1 = require("child_process");
const fs = tslib_1.__importStar(require("fs"));
const shelljs_1 = tslib_1.__importDefault(require("shelljs"));
const base_command_1 = tslib_1.__importDefault(require("../../lib/base-command"));
const path = tslib_1.__importStar(require("path"));
class New extends base_command_1.default {
    constructor() {
        super(...arguments);
        this.CURR_DIR = process.cwd();
    }
    async installModules(path) {
        this.log("installing modules");
        await this.runNpmCommand(["install", "--force", "--prefix", path]);
    }
    async formatApp(root) {
        this.log("cleaning up");
        shelljs_1.default.cd(root);
        await this.runNpmCommand(["run", "format", "--prefix", root]);
    }
    async startApp(root) {
        return new Promise((resolve, reject) => {
            const command = "npm";
            const args = ["start"];
            // Explicitly set cwd() to work around issues like
            // https://github.com/facebook/create-react-app/issues/3326.
            // Unfortunately we can only do this for Yarn because npm support for
            // equivalent --prefix flag doesn't help with this issue.
            // This is why for npm, we run checkThatNpmCanReadCwd() early instead.
            // args.push('--cwd')
            // args.push(root)
            // args.push('--verbose')
            const child = (0, child_process_1.spawn)(command, args, { stdio: "inherit", cwd: root });
            child.on("close", (code) => {
                if (code !== 0) {
                    reject({
                        command: `${command} ${args.join(" ")}`,
                    });
                    return;
                }
                resolve();
            });
        });
    }
    async cloneTemplate(projectPath) {
        this.log("Creating project in current folder.");
        if (!fs.existsSync(projectPath) && this.CURR_DIR !== projectPath) {
            fs.mkdirSync(projectPath);
        }
        shelljs_1.default.cd(projectPath);
        if (process.env.APSO_GIT_PAT) {
            shelljs_1.default.exec(`git clone --depth=1 --branch=main https://${process.env.APSO_GIT_PAT}:@github.com/mavric/apso-service-template.git ${projectPath}`);
        }
        else {
            shelljs_1.default.exec(`git clone --depth=1 --branch=main git@github.com:mavric/apso-service-template.git ${projectPath}`);
        }
        shelljs_1.default.exec(`rm -rf ${projectPath}/.git`);
        shelljs_1.default.cd(this.CURR_DIR);
    }
    async run() {
        this.log("Initializing New Apso Server...");
        const { projectName,
        // apiType,
         } = await this.validateFlags();
        const CURR_DIR = process.cwd();
        const projectPath = path.join(CURR_DIR, projectName);
        await this.cloneTemplate(projectPath);
        await this.installModules(projectPath);
        await this.formatApp(projectPath);
    }
    async validateFlags() {
        var _a;
        const { flags } = await this.parse(New);
        let projectName = flags.name;
        let apiType = (_a = flags.type) === null || _a === void 0 ? void 0 : _a.toLowerCase();
        if (typeof projectName === "undefined") {
            // this.error("`name` must be set (e.g. WoofyApp)");
            projectName = '';
        }
        if (typeof apiType === "undefined") {
            apiType = "rest";
        }
        if (apiType !== "rest" && apiType !== "graphql") {
            this.error("`apiType` must be set (e.g. rest or graphql)");
        }
        return { projectName, apiType };
    }
}
New.description = "Initialize your server project";
New.examples = [`$ apso server new --name TestProject`];
New.flags = {
    name: core_1.Flags.string({
        char: "n",
        description: "name of application",
    }),
    type: core_1.Flags.string({
        char: "t",
        description: "api type (rest or graphql)",
    }),
};
New.args = { name: core_1.Args.string() };
exports.default = New;
