// @mavric/apso-cli/dist/commands/server/scaffold.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const core_1 = require("@oclif/core");
const path = tslib_1.__importStar(require("path"));
const lib_1 = require("../../lib");
const base_command_1 = tslib_1.__importDefault(require("../../lib/base-command"));
const apsorc_parser_1 = require("../../lib/apsorc-parser");
class Scaffold extends base_command_1.default {
    async scaffoldServer(dir, entity, relationshipMap, apiType) {
        this.log(`Building... ${entity.name}`);
        const entityName = entity.name;
        const filePath = path.join(dir, entityName);
        const entityRelationships = relationshipMap[entity.name] || [];
        await (0, lib_1.createEntity)(filePath, entity, entityRelationships, apiType);
        switch (apiType) {
            case apsorc_parser_1.ApiType.Graphql:
                await this.setupGraphqlFiles(dir, entity, relationshipMap);
                break;
            case apsorc_parser_1.ApiType.Rest:
                await this.setupRestFiles(dir, entity, relationshipMap);
                break;
            default:
                break;
        }
        await (0, lib_1.createModule)(filePath, entityName, apiType);
    }
    async setupRestFiles(dir, entity, relationshipMap) {
        const filePath = path.join(dir, entity.name);
        const entityRelationships = relationshipMap[entity.name] || [];
        await (0, lib_1.createDto)(filePath, entity, entityRelationships);
        await (0, lib_1.createService)(filePath, entity.name);
        await (0, lib_1.createController)(filePath, entity, relationshipMap);
    }
    async setupGraphqlFiles(dir, entity, relationshipMap) {
        const filePath = path.join(dir, entity.name);
        const entityRelationships = relationshipMap[entity.name] || [];
        await (0, lib_1.createGqlDTO)(filePath, entity, entityRelationships);
    }
    async run() {
        const { rootFolder, entities, relationshipMap, apiType } = (0, lib_1.parseApsorc)();
        const rootPath = path.join(process.cwd(), rootFolder);
        const autogenPath = path.join(rootPath, "autogen");
        const lowerCaseApiType = apiType.toLowerCase();
        await (0, lib_1.createEnums)(autogenPath, entities, lowerCaseApiType);
        entities.forEach((entity) => {
            const scaffoldModel = this.scaffoldServer.bind(this);
            scaffoldModel(autogenPath, entity, relationshipMap, lowerCaseApiType);
        });
        await (0, lib_1.createIndexAppModule)(autogenPath, entities, lowerCaseApiType);
        await this.runNpmCommand(["run", "format"]);
    }
}
Scaffold.description = "Setup new entities and interfaces for an Apso Server";
Scaffold.examples = [`$ apso server scaffold`];
Scaffold.flags = {
    help: core_1.Flags.help({ char: "h" }),
};
Scaffold.args = {};
exports.default = Scaffold;
