// @mavric/apso-cli/dist/lib/apsorc-parser.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.parseApsorc = exports.parseApsorcV2 = exports.parseApsorcV1 = exports.ApiType = void 0;
const rc = require("rc");
const relationships_1 = require("./utils/relationships");
var ApiType;
(function (ApiType) {
    ApiType["Graphql"] = "graphql";
    ApiType["Rest"] = "rest";
})(ApiType = exports.ApiType || (exports.ApiType = {}));
const parseApsorcV1 = (apsorc) => {
    const { entities } = apsorc;
    const relationshipMap = (0, relationships_1.parseV1Relationships)(entities);
    for (const entity of entities) {
        delete entity.associations;
    }
    return { entities, relationshipMap };
};
exports.parseApsorcV1 = parseApsorcV1;
const parseApsorcV2 = (apsorc) => {
    const { entities, relationships: apsoRelationships } = apsorc;
    const relationshipMap = (0, relationships_1.parseRelationships)(apsoRelationships);
    return { entities, relationshipMap };
};
exports.parseApsorcV2 = parseApsorcV2;
const parseRc = () => {
    const apsoConfig = rc("apso");
    const rootFolder = apsoConfig.rootFolder || "src";
    const apiType = apsoConfig.apiType || "Rest";
    const version = apsoConfig.version || 1;
    const entities = apsoConfig.entities || [];
    const relationships = apsoConfig.relationships || [];
    return {
        rootFolder,
        apiType,
        version,
        entities,
        relationships,
    };
};
const parseApsorc = () => {
    const apsoConfig = parseRc();
    if (apsoConfig.version === 1 &&
        (apsoConfig === null || apsoConfig === void 0 ? void 0 : apsoConfig.apiType.toLowerCase()) !== ApiType.Rest.toLowerCase()) {
        throw new Error(`Graphql is not supported for apsorc version 1. In order to use Graphql make sure your apsorc file is version 2 compatible.`);
    }
    else {
        switch (apsoConfig.version) {
            case 1:
                return {
                    rootFolder: apsoConfig.rootFolder,
                    apiType: apsoConfig.apiType,
                    ...(0, exports.parseApsorcV1)(apsoConfig),
                };
            case 2:
                return {
                    rootFolder: apsoConfig.rootFolder,
                    apiType: apsoConfig.apiType,
                    ...(0, exports.parseApsorcV2)(apsoConfig),
                };
        }
        throw new Error(`Invalid apsorc config version: ${apsoConfig.version}`);
    }
};
exports.parseApsorc = parseApsorc;
