// @mavric/apso-cli/dist/lib/dto.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createDto = void 0;
const tslib_1 = require("tslib");
const Eta = tslib_1.__importStar(require("eta"));
const path = tslib_1.__importStar(require("path"));
const file_system_1 = require("./utils/file-system");
const field_1 = require("./utils/field");
const relationships_1 = require("./utils/relationships");
const createDto = async (apiBaseDir, entity, relationships) => {
    const { name, fields = [] } = entity;
    const Dir = path.join(apiBaseDir, "dtos");
    const File = path.join(Dir, `${name}.dto.ts`);
    const relationshipFields = relationships
        .filter((relationship) => relationship.type === "ManyToOne")
        .map((relationship) => (0, relationships_1.getRelationshipIdField)(relationship));
    const columns = [
        ...fields.map((field) => ({
            name: field.name,
            type: field.type,
            primary: field.primary,
            dataType: field.type === "enum"
                ? (0, field_1.fieldToEnumType)(field.name, name)
                : (0, field_1.getJsTypeFromFieldType)(field.type),
        })),
        ...relationshipFields.map((fieldName) => ({
            name: fieldName,
            dataType: "number",
        })),
    ];
    const primaryKeyColumns = columns.filter((column) => column.primary === true);
    const addDefaultPKProperty = primaryKeyColumns.length === 0;
    const createdAt = entity.created_at;
    const updatedAt = entity.updated_at;
    const data = {
        name: entity.name,
        addDefaultPKProperty,
        createdAt,
        updatedAt,
        columns,
        importEnums: (0, field_1.typeExistsInEntity)(entity, "enum") !== -1,
    };
    Eta.configure({
        views: path.join(__dirname, "templates"),
    });
    const content = await Eta.renderFileAsync("./rest/dto-rest", data);
    (0, file_system_1.createFile)(File, content);
};
exports.createDto = createDto;
