// @mavric/apso-cli/dist/lib/entity.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createEntity = void 0;
const tslib_1 = require("tslib");
const Eta = tslib_1.__importStar(require("eta"));
const path = tslib_1.__importStar(require("path"));
const pluralize_1 = tslib_1.__importDefault(require("pluralize"));
const file_system_1 = require("./utils/file-system");
const casing_1 = require("./utils/casing");
const field_1 = require("./utils/field");
const relationships_1 = require("./utils/relationships");
const createEntity = async (apiBaseDir, entity, relationshipsNew, apiType) => {
    const { name, fields = [], indexes, uniques } = entity;
    const File = path.join(apiBaseDir, `${name}.entity.ts`);
    const columns = (0, field_1.getFieldForTemplate)(fields, name);
    const createPrimaryKey = columns.filter((column) => column.primary === true)
        .length === 0;
    const createdAt = entity.created_at;
    const updatedAt = entity.updated_at;
    const relationships = (0, relationships_1.getRelationshipForTemplate)(name, relationshipsNew);
    const entitiesToImport = (0, relationships_1.getRelationshipsForImport)(name, relationships);
    const data = {
        name,
        indexes: indexes || [],
        uniques: uniques || [],
        createPrimaryKey,
        snakeCasedName: (0, casing_1.snakeCase)(name),
        createdAt,
        updatedAt,
        pluralizedName: (0, pluralize_1.default)((0, casing_1.camelCase)(name)),
        columns,
        associations: [...new Set(relationships)],
        entitiesToImport,
        importEnums: (0, field_1.typeExistsInEntity)(entity, "enum") !== -1,
        apiType,
    };
    Eta.configure({
        views: path.join(__dirname, "templates"),
    });
    const content = await Eta.renderFileAsync("./entities/entity", data);
    (0, file_system_1.createFile)(File, content);
};
exports.createEntity = createEntity;
