// @mavric/apso-cli/dist/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createGqlDTO = exports.createEnums = exports.createDto = exports.createIndexAppModule = exports.createModule = exports.createService = exports.createController = exports.createEntity = exports.parseApsorc = void 0;
var apsorc_parser_1 = require("./apsorc-parser");
Object.defineProperty(exports, "parseApsorc", { enumerable: true, get: function () { return apsorc_parser_1.parseApsorc; } });
var entity_1 = require("./entity");
Object.defineProperty(exports, "createEntity", { enumerable: true, get: function () { return entity_1.createEntity; } });
var controller_1 = require("./controller");
Object.defineProperty(exports, "createController", { enumerable: true, get: function () { return controller_1.createController; } });
var service_1 = require("./service");
Object.defineProperty(exports, "createService", { enumerable: true, get: function () { return service_1.createService; } });
var module_1 = require("./module");
Object.defineProperty(exports, "createModule", { enumerable: true, get: function () { return module_1.createModule; } });
var index_module_1 = require("./index-module");
Object.defineProperty(exports, "createIndexAppModule", { enumerable: true, get: function () { return index_module_1.createIndexAppModule; } });
var dto_1 = require("./dto");
Object.defineProperty(exports, "createDto", { enumerable: true, get: function () { return dto_1.createDto; } });
var enums_1 = require("./enums");
Object.defineProperty(exports, "createEnums", { enumerable: true, get: function () { return enums_1.createEnums; } });
var gql_dto_1 = require("./gql-dto");
Object.defineProperty(exports, "createGqlDTO", { enumerable: true, get: function () { return gql_dto_1.createGqlDTO; } });
