// @mavric/apso-cli/dist/tests/templates/entities/types/array-template.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const template_test_utils_1 = require("../utils/template-test-utils");
const globals_1 = require("@jest/globals");
(0, globals_1.describe)('Array Entity Template Tests', () => {
    // Setup Eta configuration once before all tests
    (0, globals_1.beforeAll)(() => {
        (0, template_test_utils_1.configureEta)();
    });
    const templatePath = './entities/entity-col-array';
    const testCases = [
        {
            name: 'should render basic text array field correctly',
            field: {
                name: 'tags',
                dataType: 'string[]',
                nullable: false
            },
            assertions: [
                { type: 'contains', text: '@Column({' },
                { type: 'contains', text: 'type: "text"' },
                { type: 'contains', text: 'array: true' },
                { type: 'contains', text: 'tags: string[];' },
                { type: 'notContains', text: 'nullable: true' }
            ]
        },
        {
            name: 'should render nullable array field correctly',
            field: {
                name: 'optionalTags',
                dataType: 'string[]',
                nullable: true
            },
            assertions: [
                { type: 'contains', text: 'nullable: true' },
                { type: 'contains', text: 'optionalTags: string[];' }
            ]
        },
        {
            name: 'should render indexed array field correctly',
            field: {
                name: 'categories',
                dataType: 'string[]',
                nullable: false,
                index: true
            },
            assertions: [
                { type: 'contains', text: '@Index()' },
                { type: 'contains', text: 'categories: string[];' }
            ]
        },
        {
            name: 'should render integer array field correctly',
            field: {
                name: 'scores',
                dataType: 'number[]',
                elementType: 'int',
                nullable: false
            },
            assertions: [
                { type: 'contains', text: 'type: "int"' },
                { type: 'contains', text: 'array: true' },
                { type: 'contains', text: 'scores: number[];' }
            ]
        },
        {
            name: 'should render decimal array field correctly',
            field: {
                name: 'amounts',
                dataType: 'number[]',
                elementType: 'decimal',
                nullable: false
            },
            assertions: [
                { type: 'contains', text: 'type: "decimal"' },
                { type: 'contains', text: 'array: true' },
                { type: 'contains', text: 'amounts: number[];' }
            ]
        }
    ];
    (0, template_test_utils_1.runTemplateTests)(templatePath, testCases);
});
