// @mavric/apso-cli/dist/tests/templates/entities/types/boolean-template.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const template_test_utils_1 = require("../utils/template-test-utils");
const globals_1 = require("@jest/globals");
(0, globals_1.describe)('Boolean Entity Template Tests', () => {
    // Setup Eta configuration once before all tests
    (0, globals_1.beforeAll)(() => {
        (0, template_test_utils_1.configureEta)();
    });
    const templatePath = './entities/entity-col-boolean';
    const testCases = [
        {
            name: 'should render basic boolean field correctly',
            field: {
                name: 'isActive',
                dataType: 'boolean',
                nullable: false
            },
            assertions: [
                { type: 'contains', text: '@IsOptional({ groups: [CREATE, UPDATE] })' },
                { type: 'contains', text: '@IsBoolean({ always: true })' },
                { type: 'contains', text: "@Column({ type: 'boolean'" },
                { type: 'contains', text: 'default: false' },
                { type: 'contains', text: 'isActive!: boolean;' },
                { type: 'notContains', text: '@Index()' },
                { type: 'notContains', text: '@PrimaryColumn()' }
            ]
        },
        {
            name: 'should render nullable boolean field correctly',
            field: {
                name: 'optionalFlag',
                dataType: 'boolean',
                nullable: true
            },
            assertions: [
                { type: 'contains', text: '@IsOptional({ groups: [CREATE, UPDATE] })' },
                { type: 'contains', text: '@IsBoolean({ always: true })' },
                { type: 'contains', text: "@Column({ type: 'boolean'" },
                { type: 'contains', text: 'optionalFlag!: boolean;' }
            ]
        },
        {
            name: 'should render boolean field with default value correctly',
            field: {
                name: 'defaultFlag',
                dataType: 'boolean',
                nullable: false,
                default: 'true'
            },
            assertions: [
                { type: 'contains', text: '@IsOptional({ groups: [CREATE, UPDATE] })' },
                { type: 'contains', text: '@IsBoolean({ always: true })' },
                { type: 'contains', text: "@Column({ type: 'boolean'" },
                { type: 'contains', text: 'default: true' },
                { type: 'contains', text: 'defaultFlag!: boolean;' }
            ]
        },
        {
            name: 'should render indexed boolean field correctly',
            field: {
                name: 'indexedFlag',
                dataType: 'boolean',
                nullable: false,
                index: true
            },
            assertions: [
                { type: 'contains', text: '@IsOptional({ groups: [CREATE, UPDATE] })' },
                { type: 'contains', text: '@IsBoolean({ always: true })' },
                { type: 'contains', text: "@Column({ type: 'boolean'" },
                { type: 'contains', text: '@Index()' },
                { type: 'contains', text: 'indexedFlag!: boolean;' }
            ]
        }
    ];
    (0, template_test_utils_1.runTemplateTests)(templatePath, testCases);
});
