// @mavric/apso-cli/dist/tests/templates/entities/types/bytea-template.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const template_test_utils_1 = require("../utils/template-test-utils");
const globals_1 = require("@jest/globals");
(0, globals_1.describe)('Bytea Entity Template Tests', () => {
    // Setup Eta configuration once before all tests
    (0, globals_1.beforeAll)(() => {
        (0, template_test_utils_1.configureEta)();
    });
    const templatePath = './entities/entity-col-bytea';
    const testCases = [
        {
            name: 'should render basic bytea field correctly',
            field: {
                name: 'binaryData',
                dataType: 'Buffer',
                nullable: false
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "bytea"' },
                { type: 'contains', text: 'binaryData: Buffer;' },
                { type: 'notContains', text: '@Index()' },
                { type: 'notContains', text: '@PrimaryColumn()' }
            ]
        },
        {
            name: 'should render nullable bytea field correctly',
            field: {
                name: 'optionalData',
                dataType: 'Buffer',
                nullable: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "bytea"' },
                { type: 'contains', text: 'optionalData: Buffer;' }
            ]
        },
        {
            name: 'should render indexed bytea field correctly',
            field: {
                name: 'indexedData',
                dataType: 'Buffer',
                nullable: false,
                index: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "bytea"' },
                { type: 'contains', text: '@Index()' },
                { type: 'contains', text: 'indexedData: Buffer;' }
            ]
        },
        {
            name: 'should render primary key bytea field correctly',
            field: {
                name: 'id',
                dataType: 'Buffer',
                nullable: false,
                primary: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "bytea"' },
                { type: 'contains', text: '@PrimaryColumn()' },
                { type: 'contains', text: 'id: Buffer;' }
            ]
        }
    ];
    (0, template_test_utils_1.runTemplateTests)(templatePath, testCases);
});
