// @mavric/apso-cli/dist/tests/templates/entities/types/char-template.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const template_test_utils_1 = require("../utils/template-test-utils");
const globals_1 = require("@jest/globals");
(0, globals_1.describe)('Char Entity Template Tests', () => {
    // Setup Eta configuration once before all tests
    (0, globals_1.beforeAll)(() => {
        (0, template_test_utils_1.configureEta)();
    });
    const templatePath = './entities/entity-col-char';
    const testCases = [
        {
            name: 'should render basic char field correctly',
            field: {
                name: 'charCode',
                dataType: 'string',
                nullable: false
            },
            assertions: [
                { type: 'contains', text: '@Column({ type: "char", length: 1' },
                { type: 'contains', text: 'charCode: string;' },
                { type: 'notContains', text: '@Index()' },
                { type: 'notContains', text: '@PrimaryColumn()' }
            ]
        },
        {
            name: 'should render nullable char field correctly',
            field: {
                name: 'optionalChar',
                dataType: 'string',
                nullable: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ type: "char"' },
                { type: 'contains', text: 'nullable: true' },
                { type: 'contains', text: 'optionalChar: string;' }
            ]
        },
        {
            name: 'should render char field with custom length correctly',
            field: {
                name: 'fixedCode',
                dataType: 'string',
                nullable: false,
                length: 5
            },
            assertions: [
                { type: 'contains', text: '@Column({ type: "char", length: 5' },
                { type: 'contains', text: 'fixedCode: string;' }
            ]
        },
        {
            name: 'should render indexed char field correctly',
            field: {
                name: 'indexedChar',
                dataType: 'string',
                nullable: false,
                index: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ type: "char"' },
                { type: 'contains', text: '@Index()' },
                { type: 'contains', text: 'indexedChar: string;' }
            ]
        }
    ];
    (0, template_test_utils_1.runTemplateTests)(templatePath, testCases);
});
