// @mavric/apso-cli/dist/tests/templates/entities/types/date-template.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const template_test_utils_1 = require("../utils/template-test-utils");
const globals_1 = require("@jest/globals");
(0, globals_1.describe)('Date Entity Template Tests', () => {
    // Setup Eta configuration once before all tests
    (0, globals_1.beforeAll)(() => {
        (0, template_test_utils_1.configureEta)();
    });
    const templatePath = './entities/entity-col-date';
    const testCases = [
        {
            name: 'should render basic date field correctly',
            field: {
                name: 'birthDate',
                dataType: 'Date',
                nullable: false
            },
            assertions: [
                { type: 'contains', text: "@Column({ type: 'date'" },
                { type: 'contains', text: 'nullable: false' },
                { type: 'contains', text: 'birthDate: string;' },
                { type: 'notContains', text: '@Index()' }
            ]
        },
        {
            name: 'should render nullable date field correctly',
            field: {
                name: 'optionalDate',
                dataType: 'Date',
                nullable: true
            },
            assertions: [
                { type: 'contains', text: "@Column({ type: 'date'" },
                { type: 'contains', text: 'nullable: true' },
                { type: 'contains', text: 'optionalDate: string;' }
            ]
        },
        {
            name: 'should render indexed date field correctly',
            field: {
                name: 'indexedDate',
                dataType: 'Date',
                nullable: false,
                index: true
            },
            assertions: [
                { type: 'contains', text: "@Column({ type: 'date'" },
                { type: 'contains', text: 'nullable: false' },
                { type: 'contains', text: '@Index()' },
                { type: 'contains', text: 'indexedDate: string;' }
            ]
        }
    ];
    (0, template_test_utils_1.runTemplateTests)(templatePath, testCases);
});
