// @mavric/apso-cli/dist/tests/templates/entities/types/decimal-template.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const template_test_utils_1 = require("../utils/template-test-utils");
const globals_1 = require("@jest/globals");
(0, globals_1.describe)('Decimal Entity Template Tests', () => {
    // Setup Eta configuration once before all tests
    (0, globals_1.beforeAll)(() => {
        (0, template_test_utils_1.configureEta)();
    });
    const templatePath = './entities/entity-col-decimal';
    const testCases = [
        {
            name: 'should render basic decimal field correctly',
            field: {
                name: 'price',
                dataType: 'number',
                nullable: false
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "decimal"' },
                { type: 'contains', text: 'precision: 10' },
                { type: 'contains', text: 'scale: 2' },
                { type: 'contains', text: 'price: number;' },
                { type: 'notContains', text: '@Index()' },
                { type: 'notContains', text: '@PrimaryColumn()' }
            ]
        },
        {
            name: 'should render nullable decimal field correctly',
            field: {
                name: 'optionalPrice',
                dataType: 'number',
                nullable: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "decimal"' },
                { type: 'contains', text: 'optionalPrice: number;' }
            ]
        },
        {
            name: 'should render decimal field with custom precision and scale',
            field: {
                name: 'amount',
                dataType: 'number',
                nullable: false,
                precision: 16,
                scale: 6
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "decimal"' },
                { type: 'contains', text: 'precision: 16' },
                { type: 'contains', text: 'scale: 6' },
                { type: 'contains', text: 'amount: number;' }
            ]
        },
        {
            name: 'should render decimal field with default value correctly',
            field: {
                name: 'defaultPrice',
                dataType: 'number',
                nullable: false,
                default: '0.00'
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "decimal"' },
                { type: 'contains', text: 'default:  0.00' },
                { type: 'contains', text: 'defaultPrice: number;' }
            ]
        },
        {
            name: 'should render indexed decimal field correctly',
            field: {
                name: 'indexedPrice',
                dataType: 'number',
                nullable: false,
                index: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "decimal"' },
                { type: 'contains', text: '@Index()' },
                { type: 'contains', text: 'indexedPrice: number;' }
            ]
        }
    ];
    (0, template_test_utils_1.runTemplateTests)(templatePath, testCases);
});
