// @mavric/apso-cli/dist/tests/templates/entities/types/enum-template.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const template_test_utils_1 = require("../utils/template-test-utils");
const globals_1 = require("@jest/globals");
(0, globals_1.describe)('Enum Entity Template Tests', () => {
    // Setup Eta configuration once before all tests
    (0, globals_1.beforeAll)(() => {
        (0, template_test_utils_1.configureEta)();
    });
    const templatePath = './entities/entity-col-enum';
    const testCases = [
        {
            name: 'should render basic enum field correctly',
            field: {
                name: 'status',
                dataType: 'enum',
                nullable: false,
                // eslint-disable-next-line camelcase
                enum_name: 'UserStatus',
                // eslint-disable-next-line camelcase
                enum_values: ['ACTIVE', 'INACTIVE', 'PENDING']
            },
            assertions: [
                { type: 'contains', text: "type: 'enum'" },
                { type: 'contains', text: 'enum: enums.enum' },
                { type: 'contains', text: 'status!: enums.enum;' },
                { type: 'notContains', text: '@Index()' },
                { type: 'notContains', text: '@PrimaryColumn()' }
            ]
        },
        {
            name: 'should render nullable enum field correctly',
            field: {
                name: 'optionalStatus',
                dataType: 'enum',
                nullable: true,
                // eslint-disable-next-line camelcase
                enum_name: 'UserStatus',
                // eslint-disable-next-line camelcase
                enum_values: ['ACTIVE', 'INACTIVE', 'PENDING']
            },
            assertions: [
                { type: 'contains', text: "type: 'enum'" },
                { type: 'contains', text: 'optionalStatus!: enums.enum;' }
            ]
        },
        {
            name: 'should render enum field with default value correctly',
            field: {
                name: 'defaultStatus',
                dataType: 'enum',
                nullable: false,
                // eslint-disable-next-line camelcase
                enum_name: 'UserStatus',
                // eslint-disable-next-line camelcase
                enum_values: ['ACTIVE', 'INACTIVE', 'PENDING'],
                default: 'UserStatus.INACTIVE'
            },
            assertions: [
                { type: 'contains', text: "type: 'enum'" },
                { type: 'contains', text: 'default: enums.enum.UserStatus.INACTIVE' },
                { type: 'contains', text: 'defaultStatus!: enums.enum;' }
            ]
        },
        {
            name: 'should render indexed enum field correctly',
            field: {
                name: 'indexedStatus',
                dataType: 'enum',
                nullable: false,
                // eslint-disable-next-line camelcase
                enum_name: 'UserStatus',
                // eslint-disable-next-line camelcase
                enum_values: ['ACTIVE', 'INACTIVE', 'PENDING'],
                index: true
            },
            assertions: [
                { type: 'contains', text: "type: 'enum'" },
                { type: 'contains', text: '@Index()' },
                { type: 'contains', text: 'indexedStatus!: enums.enum;' }
            ]
        }
    ];
    (0, template_test_utils_1.runTemplateTests)(templatePath, testCases);
});
