// @mavric/apso-cli/dist/tests/templates/entities/types/inet-template.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const template_test_utils_1 = require("../utils/template-test-utils");
const globals_1 = require("@jest/globals");
(0, globals_1.describe)('Inet Entity Template Tests', () => {
    // Setup Eta configuration once before all tests
    (0, globals_1.beforeAll)(() => {
        (0, template_test_utils_1.configureEta)();
    });
    const templatePath = './entities/entity-col-inet';
    const testCases = [
        {
            name: 'should render basic inet field correctly',
            field: {
                name: 'ipAddress',
                dataType: 'string',
                nullable: false
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "inet"' },
                { type: 'contains', text: 'ipAddress: string;' },
                { type: 'notContains', text: '@Index()' },
                { type: 'notContains', text: '@PrimaryColumn()' }
            ]
        },
        {
            name: 'should render nullable inet field correctly',
            field: {
                name: 'optionalIp',
                dataType: 'string',
                nullable: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "inet"' },
                { type: 'contains', text: 'nullable: true' },
                { type: 'contains', text: 'optionalIp: string;' }
            ]
        },
        {
            name: 'should render inet field with default value correctly',
            field: {
                name: 'defaultIp',
                dataType: 'string',
                nullable: false,
                default: '127.0.0.1'
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "inet"' },
                { type: 'contains', text: 'default:  127.0.0.1' },
                { type: 'contains', text: 'defaultIp: string;' }
            ]
        },
        {
            name: 'should render indexed inet field correctly',
            field: {
                name: 'indexedIp',
                dataType: 'string',
                nullable: false,
                index: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "inet"' },
                { type: 'contains', text: '@Index()' },
                { type: 'contains', text: 'indexedIp: string;' }
            ]
        }
    ];
    (0, template_test_utils_1.runTemplateTests)(templatePath, testCases);
});
