// @mavric/apso-cli/dist/tests/templates/entities/types/interval-template.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const template_test_utils_1 = require("../utils/template-test-utils");
const globals_1 = require("@jest/globals");
(0, globals_1.describe)('Interval Entity Template Tests', () => {
    // Setup Eta configuration once before all tests
    (0, globals_1.beforeAll)(() => {
        (0, template_test_utils_1.configureEta)();
    });
    const templatePath = './entities/entity-col-interval';
    const testCases = [
        {
            name: 'should render basic interval field correctly',
            field: {
                name: 'duration',
                dataType: 'string',
                nullable: false
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "interval"' },
                { type: 'contains', text: 'duration: string;' },
                { type: 'notContains', text: '@Index()' },
                { type: 'notContains', text: '@PrimaryColumn()' }
            ]
        },
        {
            name: 'should render nullable interval field correctly',
            field: {
                name: 'optionalDuration',
                dataType: 'string',
                nullable: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "interval"' },
                { type: 'contains', text: 'optionalDuration: string;' }
            ]
        },
        {
            name: 'should render interval field with default value correctly',
            field: {
                name: 'defaultDuration',
                dataType: 'string',
                nullable: false,
                default: '1 day'
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "interval"' },
                { type: 'contains', text: 'default:  1 day' },
                { type: 'contains', text: 'defaultDuration: string;' }
            ]
        },
        {
            name: 'should render indexed interval field correctly',
            field: {
                name: 'indexedDuration',
                dataType: 'string',
                nullable: false,
                index: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "interval"' },
                { type: 'contains', text: '@Index()' },
                { type: 'contains', text: 'indexedDuration: string;' }
            ]
        }
    ];
    (0, template_test_utils_1.runTemplateTests)(templatePath, testCases);
});
