// @mavric/apso-cli/dist/tests/templates/entities/types/serial-template.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const template_test_utils_1 = require("../utils/template-test-utils");
const globals_1 = require("@jest/globals");
(0, globals_1.describe)('Serial Entity Template Tests', () => {
    // Setup Eta configuration once before all tests
    (0, globals_1.beforeAll)(() => {
        (0, template_test_utils_1.configureEta)();
    });
    (0, globals_1.describe)('serial', () => {
        const templatePath = './entities/entity-col-serial';
        const testCases = [
            {
                name: 'should render basic serial field correctly',
                field: {
                    name: 'id',
                    dataType: 'number',
                    nullable: false
                },
                assertions: [
                    { type: 'contains', text: '@Column({ "type": "int" })' },
                    { type: 'contains', text: '@Generated("increment")' },
                    { type: 'contains', text: 'id: number;' },
                    { type: 'notContains', text: 'default:' }
                ]
            },
            {
                name: 'should render indexed serial field correctly',
                field: {
                    name: 'recordId',
                    dataType: 'number',
                    nullable: false,
                    index: true
                },
                assertions: [
                    { type: 'contains', text: '@Column({ "type": "int" })' },
                    { type: 'contains', text: '@Generated("increment")' },
                    { type: 'contains', text: '@Index()' },
                    { type: 'contains', text: 'recordId: number;' }
                ]
            },
            {
                name: 'should render primary key serial field correctly',
                field: {
                    name: 'id',
                    dataType: 'number',
                    nullable: false,
                    primary: true
                },
                assertions: [
                    { type: 'contains', text: '@Column({ "type": "int" })' },
                    { type: 'contains', text: '@Generated("increment")' },
                    { type: 'contains', text: '@PrimaryColumn()' },
                    { type: 'contains', text: 'id: number;' }
                ]
            }
        ];
        (0, template_test_utils_1.runTemplateTests)(templatePath, testCases);
    });
    (0, globals_1.describe)('bigserial', () => {
        const templatePath = './entities/entity-col-bigserial';
        const testCases = [
            {
                name: 'should render basic bigserial field correctly',
                field: {
                    name: 'id',
                    dataType: 'string',
                    nullable: false
                },
                assertions: [
                    { type: 'contains', text: '@Column({ "type": "bigint" })' },
                    { type: 'contains', text: '@Generated("increment")' },
                    { type: 'contains', text: 'id: string;' }
                ]
            },
            {
                name: 'should render primary key bigserial field correctly',
                field: {
                    name: 'id',
                    dataType: 'string',
                    nullable: false,
                    primary: true
                },
                assertions: [
                    { type: 'contains', text: '@Column({ "type": "bigint" })' },
                    { type: 'contains', text: '@Generated("increment")' },
                    { type: 'contains', text: '@PrimaryColumn()' },
                    { type: 'contains', text: 'id: string;' }
                ]
            }
        ];
        (0, template_test_utils_1.runTemplateTests)(templatePath, testCases);
    });
    (0, globals_1.describe)('smallserial', () => {
        const templatePath = './entities/entity-col-smallserial';
        const testCases = [
            {
                name: 'should render basic smallserial field correctly',
                field: {
                    name: 'id',
                    dataType: 'number',
                    nullable: false
                },
                assertions: [
                    { type: 'contains', text: '@Column({ "type": "smallint" })' },
                    { type: 'contains', text: '@Generated("increment")' },
                    { type: 'contains', text: 'id: number;' }
                ]
            }
        ];
        (0, template_test_utils_1.runTemplateTests)(templatePath, testCases);
    });
});
