// @mavric/apso-cli/dist/tests/templates/entities/types/smallint-template.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const template_test_utils_1 = require("../utils/template-test-utils");
const globals_1 = require("@jest/globals");
(0, globals_1.describe)('SmallInt Entity Template Tests', () => {
    // Setup Eta configuration once before all tests
    (0, globals_1.beforeAll)(() => {
        (0, template_test_utils_1.configureEta)();
    });
    const templatePath = './entities/entity-col-smallint';
    const testCases = [
        {
            name: 'should render basic smallint field correctly',
            field: {
                name: 'smallCounter',
                dataType: 'number',
                nullable: false
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "smallint"' },
                { type: 'contains', text: 'smallCounter: number;' },
                { type: 'notContains', text: '@Index()' },
                { type: 'notContains', text: '@PrimaryColumn()' }
            ]
        },
        {
            name: 'should render nullable smallint field correctly',
            field: {
                name: 'optionalCounter',
                dataType: 'number',
                nullable: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "smallint"' },
                { type: 'contains', text: 'optionalCounter: number;' }
            ]
        },
        {
            name: 'should render smallint field with default value correctly',
            field: {
                name: 'defaultCounter',
                dataType: 'number',
                nullable: false,
                default: 0
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "smallint"' },
                { type: 'contains', text: 'default:  0' },
                { type: 'contains', text: 'defaultCounter: number;' }
            ]
        },
        {
            name: 'should render indexed smallint field correctly',
            field: {
                name: 'indexedCounter',
                dataType: 'number',
                nullable: false,
                index: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "smallint"' },
                { type: 'contains', text: '@Index()' },
                { type: 'contains', text: 'indexedCounter: number;' }
            ]
        },
        {
            name: 'should render primary key smallint field correctly',
            field: {
                name: 'id',
                dataType: 'number',
                nullable: false,
                primary: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "smallint"' },
                { type: 'contains', text: '@PrimaryColumn()' },
                { type: 'contains', text: 'id: number;' }
            ]
        }
    ];
    (0, template_test_utils_1.runTemplateTests)(templatePath, testCases);
});
