// @mavric/apso-cli/dist/tests/templates/entities/types/text-template.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const template_test_utils_1 = require("../utils/template-test-utils");
const globals_1 = require("@jest/globals");
(0, globals_1.describe)('Text Entity Template Tests', () => {
    // Setup Eta configuration once before all tests
    (0, globals_1.beforeAll)(() => {
        (0, template_test_utils_1.configureEta)();
    });
    const templatePath = './entities/entity-col-text';
    const testCases = [
        {
            name: 'should render basic text field correctly',
            field: {
                name: 'content',
                dataType: 'string',
                nullable: false
            },
            assertions: [
                { type: 'contains', text: '@Column({ type: "text"' },
                { type: 'contains', text: '@IsOptional({ groups: [UPDATE] })' },
                { type: 'contains', text: '@IsNotEmpty({ groups: [CREATE] })' },
                { type: 'contains', text: 'content: string;' },
                { type: 'notContains', text: '@Index()' }
            ]
        },
        {
            name: 'should render nullable text field correctly',
            field: {
                name: 'optionalContent',
                dataType: 'string',
                nullable: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ type: "text"' },
                { type: 'contains', text: 'nullable: true' },
                { type: 'contains', text: '@IsOptional({ groups: [UPDATE] })' },
                { type: 'contains', text: 'optionalContent: string;' }
            ]
        },
        {
            name: 'should render text field with default value correctly',
            field: {
                name: 'defaultContent',
                dataType: 'string',
                nullable: false,
                default: "'empty'"
            },
            assertions: [
                { type: 'contains', text: '@Column({ type: "text"' },
                { type: 'contains', text: 'defaultContent: string;' }
            ]
        },
        {
            name: 'should render indexed text field correctly',
            field: {
                name: 'indexedContent',
                dataType: 'string',
                nullable: false,
                index: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ type: "text"' },
                { type: 'contains', text: '@Index()' },
                { type: 'contains', text: 'indexedContent: string;' }
            ]
        }
    ];
    (0, template_test_utils_1.runTemplateTests)(templatePath, testCases);
});
