// @mavric/apso-cli/dist/tests/templates/entities/types/tsvector-template.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const template_test_utils_1 = require("../utils/template-test-utils");
const globals_1 = require("@jest/globals");
(0, globals_1.describe)('TSVector Entity Template Tests', () => {
    // Setup Eta configuration once before all tests
    (0, globals_1.beforeAll)(() => {
        (0, template_test_utils_1.configureEta)();
    });
    const templatePath = './entities/entity-col-tsvector';
    const testCases = [
        {
            name: 'should render basic tsvector field correctly',
            field: {
                name: 'searchVector',
                dataType: 'string',
                nullable: false
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "tsvector"' },
                { type: 'contains', text: 'searchVector: string;' },
                { type: 'notContains', text: '@Index()' },
                { type: 'notContains', text: '@PrimaryColumn()' }
            ]
        },
        {
            name: 'should render nullable tsvector field correctly',
            field: {
                name: 'optionalVector',
                dataType: 'string',
                nullable: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "tsvector"' },
                { type: 'contains', text: 'nullable: true' },
                { type: 'contains', text: 'optionalVector: string;' }
            ]
        },
        {
            name: 'should render indexed tsvector field correctly',
            field: {
                name: 'indexedVector',
                dataType: 'string',
                nullable: false,
                index: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "tsvector"' },
                { type: 'contains', text: '@Index()' },
                { type: 'contains', text: 'indexedVector: string;' }
            ]
        }
    ];
    (0, template_test_utils_1.runTemplateTests)(templatePath, testCases);
});
