// @mavric/apso-cli/dist/tests/templates/entities/types/uuid-template.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const template_test_utils_1 = require("../utils/template-test-utils");
const globals_1 = require("@jest/globals");
(0, globals_1.describe)('UUID Entity Template Tests', () => {
    // Setup Eta configuration once before all tests
    (0, globals_1.beforeAll)(() => {
        (0, template_test_utils_1.configureEta)();
    });
    const templatePath = './entities/entity-col-uuid';
    const testCases = [
        {
            name: 'should render basic uuid field correctly',
            field: {
                name: 'id',
                dataType: 'string',
                nullable: false
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "uuid"' },
                { type: 'contains', text: 'id: string;' },
                { type: 'notContains', text: '@Index()' },
                { type: 'notContains', text: '@PrimaryColumn()' }
            ]
        },
        {
            name: 'should render nullable uuid field correctly',
            field: {
                name: 'optionalId',
                dataType: 'string',
                nullable: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "uuid"' },
                { type: 'contains', text: 'optionalId: string;' }
            ]
        },
        {
            name: 'should render uuid field with default value correctly',
            field: {
                name: 'defaultId',
                dataType: 'string',
                nullable: false,
                default: 'uuid_generate_v4()'
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "uuid"' },
                { type: 'contains', text: 'default:  uuid_generate_v4()' },
                { type: 'contains', text: 'defaultId: string;' }
            ]
        },
        {
            name: 'should render indexed uuid field correctly',
            field: {
                name: 'indexedId',
                dataType: 'string',
                nullable: false,
                index: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "uuid"' },
                { type: 'contains', text: '@Index()' },
                { type: 'contains', text: 'indexedId: string;' }
            ]
        },
        {
            name: 'should render primary key uuid field correctly',
            field: {
                name: 'primaryId',
                dataType: 'string',
                nullable: false,
                primary: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "uuid"' },
                { type: 'contains', text: '@PrimaryColumn()' },
                { type: 'contains', text: 'primaryId: string;' }
            ]
        }
    ];
    (0, template_test_utils_1.runTemplateTests)(templatePath, testCases);
});
