// @mavric/apso-cli/dist/tests/templates/entities/types/xml-template.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const template_test_utils_1 = require("../utils/template-test-utils");
const globals_1 = require("@jest/globals");
(0, globals_1.describe)('XML Entity Template Tests', () => {
    // Setup Eta configuration once before all tests
    (0, globals_1.beforeAll)(() => {
        (0, template_test_utils_1.configureEta)();
    });
    const templatePath = './entities/entity-col-xml';
    const testCases = [
        {
            name: 'should render basic xml field correctly',
            field: {
                name: 'xmlContent',
                dataType: 'string',
                nullable: false
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "xml"' },
                { type: 'contains', text: 'xmlContent: string;' },
                { type: 'notContains', text: '@Index()' },
                { type: 'notContains', text: '@PrimaryColumn()' }
            ]
        },
        {
            name: 'should render nullable xml field correctly',
            field: {
                name: 'optionalXml',
                dataType: 'string',
                nullable: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "xml"' },
                { type: 'contains', text: 'nullable: true' },
                { type: 'contains', text: 'optionalXml: string;' }
            ]
        },
        {
            name: 'should render indexed xml field correctly',
            field: {
                name: 'indexedXml',
                dataType: 'string',
                nullable: false,
                index: true
            },
            assertions: [
                { type: 'contains', text: '@Column({ "type": "xml"' },
                { type: 'contains', text: '@Index()' },
                { type: 'contains', text: 'indexedXml: string;' }
            ]
        }
    ];
    (0, template_test_utils_1.runTemplateTests)(templatePath, testCases);
});
